node/state/core.go

```diff
@@ -31,9 +33,39 @@ func CoreAccessor(endpoint string) func(fx.Lifecycle, keystore.Keystore, params.
 			return nil, err
 		}
 		signer := apptypes.NewKeyringSigner(ring, keyringAccName, string(net))
+		keys, err := signer.List()
+		if err != nil {
+			return nil, err
+		}
+		// check if key exists with prefix `celes`
+		var (
+			exists = false
+			name   = ""
+		)
+		for _, key := range keys {
+			if key.GetName()[:5] == keyringAccName {
```

Comment:
Okay how about this flow @liamsi @adlerjohn: 

* expose `--keyring.AccName` flag
* If user sets this flag during `init` or `start`, keyring constructor will try and find key prefixed by the given name, and uses it as the node's default key. If the keyring constructor cannot find a key prefixed by the given name, it will **fail out**.
* If the flag is **not set**, the default keyringAccName will be `""` such that the keyring constructor will look for keys in the keystore path. 
  * If **no keys are found**, constructor will construct a fresh new key and log that it has done so. 
  * If **several keys are found**, constructor will use the first key from the key list `keys[0]` and will use that key as the node's default account. (It will always be logged which key name the node is using as default key for node's account).
  * If one key is found, constructor will use it.

Reply:
How can I create a key with the prefix? If the only two options are fail if no key found with the given prefix or create a key with an empty prefix, then I can't ever create a key with the given prefix.

Reply:
you can do `celestia <node_type> keys add <name_of_key> --keyring-backend test`

and then `celestia <node_type> <init || start> --keyring.AccName <name_of_key>`

If the key is created for user, the name will be `my_celes_key`.